packages/driver/src/cy/multi-domain/index.ts

```diff
@@ -84,16 +87,12 @@ export function addCommands (Commands, Cypress: Cypress.Cypress, cy: Cypress.cy,
         const _reject = (err) => {
           cleanup()
           log.error(err)
-          if (typeof err === 'object') {
-            err.onFail = () => {}
-          }
-
           reject(err)
         }
 
-        const onQueueFinished = ({ err, subject, unserializableSubjectType }) => {
-          if (err) {
-            return _reject(err)
+        const onQueueFinished = ({ err, subject, unserializableSubjectType, hasMultiDomainError }) => {
+          if (hasMultiDomainError) {
+            return _reject(reifyCrossDomainError(err, userInvocationStack))
```

Comment:
Since the `switchToDomain` command itself is rejected, the userInvocationStack should be attached automatically. That's how it works for every other command. I'm curious why it's not working in some cases.

Reply:
It looks like the `userInvocationStack` is only attached when the error is a chai assertion error, but commands errors do not contain `userInvocationStack`. From what I can see, command errors get the `userInvocationStack` from `current`. This works the same for `switchToDomain` as far as I can tell.

The issue here isn't so much with finding the `userInvocationStack` when the command is rejected, but with correcting the stack to come from `switchToDomain` as opposed to the `communicator` event emitter. We only need the `userInvocationStack` for this specifically, and once the stack is corrected, the `codeFrame` can be built properly in the `fail` handler inside of `cy.ts`.

##### without reify
![](https://user-images.githubusercontent.com/3980464/157697949-8ed025df-f2c9-451b-b2c9-bc489f27062e.png)

##### with reify
![](https://user-images.githubusercontent.com/3980464/157697974-3e115e19-ebea-48d7-a4c4-75d06138be2c.png)

Reply:
We could pull user invocation from `current`, but I feel like there are cases where the `switchToDomain` CB finishes, then a command fails in the secondary and `current` has moved on from `switchToDomain`. That's the only reason it is inside a closure.

Also right now we aren't reifying any cross origin uncaught errors, which I believe fails the current command. Since they are thrown on the spec window, I don't think we have to 😄 

![](https://user-images.githubusercontent.com/3980464/157701927-ff9ee812-da06-4f07-b1e7-7839eb475271.png)